### PR TITLE
fix security issues

### PR DESCRIPTION
**Pull request by zhijian-pro**

### CODE REVIEW

**Comment by zhijian-pro** [comment]
fix https://github.com/juicedata/juicefs/security/code-scanning

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -99,3 +99,7 @@ func GuessMimeType(key string) string {
 	}
 	return mimeType
 }
+
+func EncodeUserInput(input string) string {
```
Comment (davies) [inline pkg/utils/utils.go:103]: RemoveNewLine().

It seems that this is only used by pkg/fs, we should put it there.

File: pkg/fs/http.go [hunk 2]
```diff
@@ -127,7 +128,7 @@ func (hfs *webdavFS) Rename(ctx context.Context, oldName, newName string) error
 }
 
 func (hfs *webdavFS) Stat(ctx context.Context, name string) (os.FileInfo, error) {
-	fi, err := hfs.fs.Stat(hfs.ctx, name)
+	fi, err := hfs.fs.Stat(hfs.ctx, utils.EncodeUserInput(name))
```
Comment (SandyXSD) [inline pkg/fs/http.go:131]: Is it necessary? If yes, should we also apply it on other methods?

Comment (SandyXSD) [reply pkg/fs/http.go]: @zhijian-pro no alert for this line

Comment (zhijian-pro) [reply pkg/fs/http.go]: image
Here is the source of several other alerts.

**Review by SandyXSD — APPROVED** [review]

**Pull request merged**